video/out/vo.h

```diff
@@ -224,6 +226,21 @@ struct vo_frame {
     uint64_t frame_id;
 };
 
+struct vo_buffer_data {
+    /* Inputs */
+    struct mp_image_params *par;     /* Partial, only width, height and imgfmt used */
+    int aligned_width;                          /* From avcodec_align_dimensions2() */
+    int aligned_height;                         /* From avcodec_align_dimensions2() */
+    int stride_alignment[MP_MAX_PLANES];        /* From avcodec_align_dimensions2() */
+
+    /* Outputs */
+    uint8_t *data[MP_MAX_PLANES];                             /* Pointers to planes */
+    size_t linesize[MP_MAX_PLANES];                              /* Stride in bytes */
+    size_t total_size;                     /* Total size of the buffer with padding */
+    void *opaque;                            /* Context to give to release function */
+    void (*release_buffer)(void *opaque, uint8_t *data);        /* Release function */
```

Comment:
So I don't understand why you don't pass a mp_image reference to the VO, and let it set a refcounted buffer.